src/Microsoft.Framework.PackageManager/Building/BuildManager.cs

```diff
@@ -34,9 +41,63 @@ public BuildManager(IServiceProvider hostServices, BuildOptions buildOptions)
 
         public bool Build()
         {
+            var projectFilesFinder = new Matcher();
+
+            // Resolve all the project names
+            var projectFilesToBuild = new List<string>();
+            foreach(var pattern in _buildOptions.ProjectPatterns)
+            {
+                if (pattern.Contains("*"))
+                {
+                    // Requires globbing
+                    projectFilesFinder.AddInclude(NormalizeGlobbingPattern(pattern));
+                }
+                else
+                {
+                    projectFilesToBuild.Add(pattern);
+                }
+            }
+
+            var rootDirectory = Directory.GetCurrentDirectory();
+            var patternSearchFolder = new DirectoryInfoWrapper(new DirectoryInfo(rootDirectory));
+            var globbingProjects = projectFilesFinder.Execute(patternSearchFolder).Files.Select(file =>
+                Path.Combine(rootDirectory, file));
+            projectFilesToBuild.AddRange(globbingProjects);
+
+            var sw = Stopwatch.StartNew();
+
+            _cacheContextAccessor = new CacheContextAccessor();
+            _cache = new Cache(_cacheContextAccessor);
+
+            var globalSucess = true;
+            foreach (var project in projectFilesToBuild)
```

Comment:
That's not a bad suggestion, it could make build even faster when there are a lot of projects but I think that's part of a much bigger change. I want to keep this change as small as possible